mixer/pkg/il/evaluator/expression.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package evaluator
+
+import "istio.io/istio/mixer/pkg/attribute"
+
+// Expression represents a precompiled expression that can be immediately evaluated.
+// It holds no cache and does not listen to any events. If the configuration changes, the CompiledExpression needs
+// to be discarded and created again.
+type Expression interface {
+	// Evaluate evaluates this compiled expression against the attribute bag.
+	Evaluate(attributes attribute.Bag) (interface{}, error)
+
+	// EvaluateBoolean evaluates this compiled expression against the attribute bag and returns the result as boolean.
+	// panics if the expression does not return a boolean.
+	EvaluateBoolean(attributes attribute.Bag) (bool, error)
```

Comment:
I presume this will blossom into EvaluateXXX for all supported types?

Reply:
Correct.